### PR TITLE
Generic external container for MCTruth

### PR DESCRIPTION
**Pull request by sawenzel**
* First version of an associative container managing MClabels for entities
  identified by some consecutive ID (such as hits, digits, clusters) sitting
  in a vector/TClonesArray.

* Container can be generalized to hold other associations than MCLabels.

### CODE REVIEW

File: DataFormats/simulation/include/SimulationDataFormat/MCTruthContainer.h [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+// to multiple TruthElements
+
+// note that we inherit from TObject just to be able to register this thing with the FairRootManager
+// (which might not be necessary in the future)
+template <typename TruthElement>
+class MCTruthContainer : public TNamed
+{
+ private:
+  std::vector<MCTruthHeaderElement>
+    mHeaderArray;                        // the header structure array serves as an index into the actual storage
```
Comment (matthiasrichter) [inline DataFormats/simulation/include/SimulationDataFormat/MCTruthContainer.h]: if you put the comment above the declaration, the member and type can be on one single line, I also suggest to start the comment with `///`, with the three slashes the comment makes it to the doxygen documentation. This is the same also for the comments before methods

**Comment by sawenzel** [comment]
Needed to progress with TPC developments. Merging. Can be refined later on.

**Comment by wiechula** [comment]
This looks fine for me. Definitely a good starting point to extend.

**Review by matthiasrichter — COMMENTED** [review]
Just a general comment regarding documentation: with three slashes `///` the comments make it into the doxygen documentation.

How about also adding some iterator functionality?

**Pull request merged**